components/api/priceProcess.js

```diff
@@ -28,7 +34,30 @@ async function findFirstEntryWithPrice(ingredient) {
     }
 }
 
-async function priceProcess(ingredientList) {
+async function priceProcess(ingredientList, db = null) {
+    console.log(db);
+    // First, attempt to get price data from FireStore
+    if (db !== null) {
+        let toDelete = [];
+
+        let promises = ingredientList.map(async (ingredient, index) => {
+            let doc = await db.collection("foods").doc(ingredient.getName()).get();
+            if (doc.exists) {
+                ingredient.setPrice(doc.data().price);
+                ingredient.setProductName(doc.data().productName);
+                toDelete.push(index);
+                return true;
+            }
+            return false;
+        });
+
+        await Promise.all(promises);
+
+        // toDelete.forEach(index => ingredientList.splice(index, 1));
+        ingredientList = ingredientList.filter((item, index) => !toDelete.includes(index));
```

Comment:
This is clean, but to technically be more algorithmically efficient you could iterate over `toDelete` in reverse order and delete freely from `ingredientList` without having to worry about deleting and shifting indexes, which is technically faster. (this is all assuming `toDeleted` is sorted)

at the moment, you have 9 operations with the following code
```
ingredientsList = [red, yellow, blue, green, black]

toDelete = [3,4]
```
when it could be 2 operations if you just deleted `ingredientsList[4]` and then `ingredientsList[3]`

Reply:
Oh, and just another comment on the theoretical runtime complexity.

Assuming the worst case that toDeleted is completely not sorted, the Big O of sorting it and then deleting would be worst case (`toDelete.length == ingredientsList.length`) **n(logn) + n**

as opposed to the identical worst case for current line 57, which would be ~ **n^2**

so it's still better to sort toDelete either way I think.

Reply:
This looks good, I will implement it.

Reply:
@joshua-yan Implemented in the new commits.